### ISSUE TITLE
[org-mode -related] [SPC SPC] does not work ([M-X] does, though)

### ISSUE THREAD

**Issue by allentiak** [syl20bnr/spacemacs#15964]
#### Reproduction guide :beetle:
- Start Emacs
- Open a file/project (in my case, a Clojure one)
- Press [SPC SPC]. This should show a list of commands, just like if you press [M-X].
- The [M-X] shortcut works fine.

*Observed behaviour:* :eyes: :broken_heart:

When pressing SPC one more time, I get "SPC SPC SPC is undefined".

*Expected behaviour:* :heart: :smile:
This should show a list of commands.

#### System Info

`~/.spacemacs.d` dir: https://github.com/allentiak/.spacemacs.d

 :computer:
- OS: gnu/linux
- Emacs: 28.2
- Spacemacs: 0.999.0
- Spacemacs branch: develop (rev. 3512e43c9)
- Graphic display: t
- Running in daemon: nil
- Distribution: spacemacs
- Editing style: vim
- Completion: helm
- Layers:
```elisp
((auto-completion :variables auto-completion-enable-help-tooltip t auto-completion-enable-snippets-in-popup t auto-completion-enable-sort-by-usage t)
 (clojure :variables cider-format-code-options
          '(("indents"
             (("ns"
               (("inner" 0)
                ("inner" 1))))))
          cider-overlays-use-font-lock t cider-preferred-build-tool 'clojure-cli cider-repl-buffer-size-limit 250 cider-result-overlay-position 'at-point clojure-backend 'lsp clojure-enable-sayid t clojure-enable-clj-refactor t clojure-enable-kaocha-runner t clojure-enable-linters
          '(clj-kondo joker)
          clojure-indent-style 'align-arguments clojure-align-forms-automatically t clojure-toplevel-inside-comment-form t)
 colors command-log dap emacs-lisp
 (git :variables git-magit-status-fullscreen t magit-diff-adjust-tab-width t magit-diff-paint-whitespace t magit-diff-highlight-trailing t magit-diff-refine-hunk 'all)
 (go :variables go-backend 'lsp go-format-before-save t go-use-gocheck-for-testing t go-use-golangci-lint t)
 helm html
 (java :variables java-backend 'lsp)
 (keyboard-layout :variables kl-layout 'colemak-neio-inverted)
 latex lsp
 (markdown :variables markdown-live-preview-engine 'vmd)
 multiple-cursors org parinfer
 (plantuml :variables plantuml-jar-path "/usr/share/plantuml/plantuml.jar" org-plantuml-jar-path "/usr/share/plantuml/plantuml.jar")
 (shell :variables shell-default-height 30 shell-default-position 'bottom)
 spacemacs-visual spell-checking sql syntax-checking
 (treemacs :variables treemacs-indentation 1 treemacs-use-filewatch-mode t treemacs-use-follow-mode t treemacs-use-git-mode 'deferred)
 (version-control :variables version-control-diff-tool 'git-gutter+ version-control-global-margin t)
 yaml)
```
- System configuration features: ACL CAIRO DBUS FREETYPE GIF GLIB GMP GNUTLS GPM GSETTINGS HARFBUZZ JPEG JSON LCMS2 LIBOTF LIBSELINUX LIBSYSTEMD LIBXML2 M17N_FLT MODULES NATIVE_COMP NOTIFY INOTIFY PDUMPER PNG RSVG SECCOMP SOUND THREADS TIFF TOOLKIT_SCROLL_BARS X11 XDBE XIM XPM GTK3 ZLIB

#### Backtrace :paw_prints:
```
<<BACKTRACE IF RELEVANT>>
```

**Comment by lebensterben** [comment]
- Try update emacs packages.
- Try checkout an earlier commit and see whether the problem persist.

**Comment by derinsh0** [comment]
It’s probably not your issue, but from my own experience when that happens to me time to time it’s because there are syntax errors in the .spacemacs file. Have you read the messages buffer or made sure spacemacs loaded OK? 

There is a bug where custom-settings can get screwy, so look at the end first.

**Comment by simonpintarelli** [comment]
I have the same problem:

```
SPC SPC- which-key: There are no keys to show
```

`M-x` still works.

Deleting  `.spacemacs` fixed it, but I haven't yet figured out what exactly causes it.

**Comment by allentiak** [comment]
@simonpintarelli 
> Deleting `.spacemacs` fixed it, but I haven't yet figured out what exactly causes it.

@simonpintarelli 
It was the same for me.

I tried modifying my `.spacemacs` file, section by section.
It seems the offending line is
https://github.com/allentiak/.spacemacs.d/blob/3f92fda881eed76f7d1410900ee464f0f34ef5f9/init.el#L320

```elisp
;; Default major mode of the scratch buffer (default `text-mode')
   dotspacemacs-scratch-mode 'org-mode
```

@simonpintarelli 
Could you try changing the aforementioned line to `'text-mode`, and seeing what happens, please?


This configuration setting was based on @practicalli-john's.

@practicalli-john I saw your current configuration repo, and you don't seem to use this setting anymore.
Did you change it because you experienced the same problem? Or you never used it in the first place and I simply mis-copied it? 


Does anyone else have the same problem?

**Comment by allentiak** [comment]
BTW: This is the full list of problems this setting triggers for me:

- the list for recent files and projects is missing at the startup screen
- line numbers are not shown when editing files
- helm does not work correctly: [SPC SPC] does not work, although [M-X] does.

**Comment by simonpintarelli** [comment]
> @simonpintarelli Could you try changing the aforementioned line to `'text-mode`, and seeing what happens, please?

I was using `org-mode` in `dotspacemacs-scratch-mode` before. In the fresh config (not fully restored, but most of the layers activated I was using before), setting scratch-mode to text-mode doesn't break anything for me.

BTW: Not sure I tried line numbers, but helm was definitely working in my case (in M-x, find-files, helm-ag, projectile find file).

**Comment by lebensterben** [comment]
@allentiak 

do you see any warning/error about org-mode?

(for example, org-asset-version not found)

**Comment by simonpintarelli** [comment]
I had the error about org-assert-version not found, but only after I threw away my old configuration. Curiously the issue about it was from January https://github.com/syl20bnr/spacemacs/issues/15869 and org-mode was still working fine for me despite the `SPC-SPC` issue. I deleted the *elc files and then it worked. However `SPC-SPC` worked despite the org error, not sure it's related.

**Comment by allentiak** [comment]
@simonpintarelli 
> setting scratch-mode to `text-mode` doesn't break anything for me.

Setting it to **`org-mode`** breaks things for me.

**Comment by allentiak** [comment]
The funny thing is... Until a few weeks ago, the same old `.spacemacs` didn't trigger this bug...

However, when I tried doing a `git bisect` on my Spacemacs repo, I couldn't reproduce the right behavior.

Maybe this could be caused by one of the latest versions of some of the packages my layers use? If so, this would make thing quite difficult to reproduce...

**Comment by practicalli-johnny** [comment]
> ```elisp
> ;; Default major mode of the scratch buffer (default `text-mode')
>    dotspacemacs-scratch-mode 'org-mode
> ```
> 
> @practicalli-john I saw your current configuration repo, and you don't seem to use this setting anymore.
> Did you change it because you experienced the same problem? Or you never used it in the first place and I simply mis-copied it? 

I was using org-mode as the `dotspacemacs-scratch-mode` until the issues with Org started occurred.

I replaced the org-mode files in elpa and initially the newly installed packages worked, but the next Emacs start up the same org issues occurred 

I rewrote the `spacemacs/init.el` from the Spacemacs temple and the only thing I didn't restore from my own config was the `dotspacemacs-scratch-mode` and `dotspacemacs-new-empty-buffer-major
-mode`

There are occasional org related warning, but nothing obviously broken

**Comment by allentiak** [comment]
@lebensterben 
> do you see any warning/error about org-mode?
> (for example, org-asset-version not found)

Yes. I've just checked the `*Messages*` buffer.

```
Package cl is deprecated
File mode specification error: (invalid-function org-assert-version)
```

I *do* have the `org` layer installed, though...

https://github.com/allentiak/.spacemacs.d/blob/154c6a50a9859e959efa404c0469adc107ef0739/init.el#L115

**Comment by allentiak** [comment]
And no, I don't know which layer could use the `cl` package...

**Comment by lebensterben** [comment]
closing in favor of #15896

**Comment by allentiak** [comment]
> closing in favor of #15896

Yeah, it seems to be an `org-mode`-related thing.

Thank you very much, @lebensterben, @derinsh, @simonpintarelli, and @practicalli-john for your help on this!
